src/NServiceBus.Core/Receiving/ProcessingCompleted.cs

```diff
@@ -0,0 +1,48 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Event raised when the transport has completed processing a message.
+    /// </summary>
+    public class ProcessingCompleted
+    {
+        /// <summary>
+        /// Constructs the event.
+        /// </summary>
+        public ProcessingCompleted(string messageId, bool wasAcknowledged, Dictionary<string, string> headers, DateTimeOffset startedAt, DateTimeOffset completedAt)
+        {
+            MessageId = messageId;
+            WasAcknowledged = wasAcknowledged;
+            Headers = headers;
+            StartedAt = startedAt;
+            CompletedAt = completedAt;
+        }
+
+        /// <summary>
+        /// The ID of the message.
+        /// </summary>
+        public string MessageId;
+
+        /// <summary>
+        /// True if the message was acknowledged and removed from the queue.
+        /// </summary>
+        public bool WasAcknowledged;
```

Comment:
I guess this is required because the event is being raised regardless of whether the message was successfully processed or not, whereas `ReceivePipelineCompleted` is only raised if the receive pipeline does not throw an exception?

Reply:
Correct